data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -108,3 +110,174 @@ def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
     )
 
     return floored_series
+
+
+def _create_df_from_yaml_contents(
+    fields_list_from_yaml: list,
+    fields_to_store_in_codebook: list,
+) -> pd.DataFrame:
+    """Helper function to create a dataframe from yaml fields
+
+    This function does:
+        1. Creates a dictionary that will eventually become a dataframe.
+           This dictionary has columns dictated by whichever fields
+           the yaml file has that we want to keep, specified by fields_to_store.
+        2. Goes through the fields and adds each type of field to the dictionary. If
+           the field is missing, appends a null value.
+        3. Returns a dataframe that includes all of the fields specified and
+           also is indexed by the column name that is native to the CEJST team
+    """
+    # this becomes the codebook frame for each  yaml source. In particular,
+    # the key becomes column names, and the lists store their values. We hard-set the
+    # first column name to be the CEJST_SCORE_COLUMN_NAME because this should be
+    # the same across the board for every component codebook.
+    codebook_dictionary = {
+        field.new_label_in_codebook: [] for field in fields_to_store_in_codebook
+    }
+    codebook_dictionary[constants.CEJST_SCORE_COLUMN_NAME] = []
+
+    # we reshape the data from a list of dictionaries to a dictionary of lists
+    # so that we can cast it as a dataframe
+    for single_field_details in fields_list_from_yaml:
+        assert constants.CEJST_SCORE_COLUMN_NAME in single_field_details, (
+            "Error: the yaml codebook should crosswalk to the native column "
+            + f"from the CEJST pipeline, called {constants.CEJST_SCORE_COLUMN_NAME}"
+        )
+        codebook_dictionary[constants.CEJST_SCORE_COLUMN_NAME].append(
+            single_field_details[constants.CEJST_SCORE_COLUMN_NAME]
+        )
+        for field_information in fields_to_store_in_codebook:
+            try:
```

Comment:
Could you add a comment about what's happening here? Why is there a try/except?

Reply:
yes